common/io/src/buffer/buffer_read_number_ext.rs

```diff
@@ -66,9 +66,17 @@ where R: BufferRead
             let _ = self.ignores(|f| (b'0'..=b'9').contains(&f))?;
         }
 
-        FromLexical::from_lexical(buf.as_slice()).map_err_to_code(ErrorCode::BadBytes, || {
-            format!("Cannot parse value:{:?} to number type", buf)
-        })
+        match buf.is_empty() {
+            true => Ok(T::default()),
```

Comment:
if empty, defaults to return default value?

Reply:
there is a setting `empty_as_default`.

Reply:
yes. string type have the same behavior.